lib/bn.js

```diff
@@ -885,13 +885,212 @@ BN.prototype._bigMulTo = function _bigMulTo(num, out) {
 
 BN.prototype.mulTo = function mulTo(num, out) {
   var res;
-  if (this.length + num.length < 63)
+  var len = this.length + num.length;
+  if (len < 63)
     res = this._smallMulTo(num, out);
-  else
+  else if (len < 1024)
     res = this._bigMulTo(num, out);
+  else
+    res = this._bigMulToF(num, out);
   return res;
 };
 
+
+// Private
+
+function FFTM(x, y) {
+  this.x = x;
+  this.y = y;
+}
+
+FFTM.prototype.makeRBT = function makeRBT(N) {
+  var t = new Array(N);
+  var l = (Math.log(N) / Math.log(2)) | 0;
+  for (var i = 0; i < N; i++) {
+    t[i] = this.revbin(i, l, N);
+  }
+
+  return t;
+};
+
+FFTM.prototype.revbin = function revbin(x, l, N) {
+  if (x === 0 || x === N - 1)
+    return x;
+
+  var rb = 0;
+  for (var i = 0; i < l; i++) {
+    rb |= (x & 1) << (l - i - 1);
+    x >>= 1;
+  }
+
+  return rb;
+};
+
+FFTM.prototype.permute = function permute(rbt, rws, iws, rtws, itws, N) {
+  for (var i = 0; i < N; i++) {
+    rtws[i] = rws[rbt[i]];
+    itws[i] = iws[rbt[i]];
+  }
+};
+
+FFTM.prototype.transform = function transform(rws, iws, rtws, itws, N, rbt) {
+  this.permute(rbt, rws, iws, rtws, itws, N);
+
+  for (var s = 1; s < N; s <<= 1) {
+    var l = s << 1;
+
+    var rtwdf = Math.cos(2 * Math.PI / l);
+    var itwdf = Math.sin(2 * Math.PI / l);
+
+    for (var p = 0; p < N; p += l) {
+      var rtwdf_ = rtwdf;
+      var itwdf_ = itwdf;
+
+      for (var j = 0; j < s; j++) {
+        var re = rtws[p + j];
+        var ie = itws[p + j];
+
+        var ro = rtws[p + j + s];
+        var io = itws[p + j + s];
+
+        var rx = rtwdf_ * ro - itwdf_ * io;
+
+        io = rtwdf_ * io + itwdf_ * ro;
+        ro = rx;
```

Comment:
Technically, `twdf_ = exp(i * PI * j / s)` here. Except that after `j` hit `l` it is `twdf_ = exp(i * PI * (j - 1) / s)`